### PR TITLE
DT-1971 Add role to retry dlq endpoint

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueResource.kt [hunk 1]
```diff
@@ -17,6 +18,7 @@ class HmppsQueueResource(private val hmppsQueueService: HmppsQueueService) {
   }
 
   @PutMapping("/retry-dlq/{dlqName}")
+  @PreAuthorize("hasRole('ROLE_QUEUE_ADMIN')")
   fun retryDlq(@PathVariable("dlqName") dlqName: String) =
```
Comment (andymarke) [inline lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueResource.kt:22]: I guess a subsequent interation will see if you can drive the role from config?

File: lib/src/test/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueResourceTest.kt [hunk 2]
```diff
@@ -10,13 +10,15 @@ import org.junit.jupiter.api.Test
...
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
 import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
 import org.springframework.boot.test.mock.mockito.MockBean
 import org.springframework.test.web.servlet.MockMvc
 import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put
 import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
 
-@WebMvcTest
+@WebMvcTest(HmppsQueueResource::class)
+@AutoConfigureMockMvc(addFilters = false)
```
Comment (andymarke) [inline lib/src/test/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueResourceTest.kt:21]: why do you need this?

**Review by andymarke — APPROVED** [review]
LGTM

File: lib/src/test/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueResourceTest.kt (hunk 2 above)
Comment (mikehalmamoj) [reply lib/src/test/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueResourceTest.kt:21]: To turn off the security filter.  WebMvcTest is fine with security when you are testing an real application but for testing a library I would need to build a real application to add security in... oh yeah I did that in the `test-app` module :smile: 

So essentially this class is a "unit" test for the resource.

**Pull request merged**